test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -0,0 +1,325 @@
+            "Setting inventory_mom for vnode_pool %s to %s" %
+            ('1', self.noninv_mom.shortname), max_attempts=5,
+            starttime=start_time)
+        if rc is None:
+            self.logger.info(
+                "Inventory mom %s is not in server logs." %
+                (self.noninv_mom.shortname))
+            self.assertTrue(rc)
+        else:
+            self.logger.info(
+                "Inventory mom is now %s in server logs." %
+                (self.noninv_mom.shortname))
+            self.assertTrue(rc)
+
+        # Check compute nodes are up
+        vlist = []
+        try:
+            vnl = self.server.filter(
+                VNODE, {'resources_available.vntype': 'cray_compute'})
+            vlist = vnl["resources_available.vntype=cray_compute"]
+        except Exception:
+            sys.exc_clear()
+
+        # Loop through each compute vnode in the list and check if state = free
+        r1 = len(vlist)
+        for i in range(0, r1):
+            v1 = vlist[i]
+            # Check that the node is in free state
+            rv = self.server.expect(
+                VNODE, {'state': 'free'}, id=v1, max_attempts=3, interval=2)
+            self.assertTrue(rv)
+
+        # Start the previous inv mom.
+        self.inv_mom.start()
+
+        # Check previous inventory mom is up
+        rv = self.server.expect(
+            VNODE, {'state': 'free'}, id=self.inv_mom.shortname,
+            max_attempts=3, interval=2)
+        self.assertTrue(rv)
+
+    def test_two_moms_different_vnode_pool(self):
+        """
+        Differing vnode_pool for two moms shall result in both moms reporting
+        inventory.
+        """
+        found_in_momA = 0
+        found_in_momB = 0
+        self.momA = self.moms.values()[0]
+        try:
+            self.momB = self.moms.values()[1]
+        except IndexError:
+            self.assertTrue(
+                False, msg="Provide at least 2 moms while invoking test")
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr_A = {'vnode_pool': '1'}
+        attr_B = {'vnode_pool': '2'}
+
+        start_time = int(time.time())
+
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA, attrib=attr_A)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB, attrib=attr_B)
+
+        rc = self.momA.log_match(
+            "Hello (no inventory required) from server", max_attempts=5,
+            starttime=start_time)
+        if rc is None:
+            found_in_momA = 0
+        else:
+            found_in_momA = 1
+
+        rc = self.momB.log_match(
+            "Hello (no inventory required) from server", max_attempts=5,
+            starttime=start_time)
+        if rc is None:
+            found_in_momB = 0
+        else:
+            found_in_momB = 1
+
+        if (found_in_momA + found_in_momB != 0):
+            self.assertTrue(False, msg="Both moms must report inventory")
```

Comment:
Put the conditional from your if into the self.assertTrue().  I'm confused by your conditional.  Won't this fail the test unless the conditional == 0?  That seems wrong to me.  Isn't the success case conditional == 2?